### PR TITLE
feat: add json schema

### LINKED ISSUE #27: feat(config): define JSON schema
See https://json-schema.org/

### PR DESCRIPTION
**Pull request by lukashass**
Closes #27 

Possibilities for further improvement:

- [ ] ~split schema into plugins~ (not needed with #40)
- [ ] ~extract schema from go types (https://github.com/alecthomas/jsonschema)~ #40 
- [x] deploy to GitHub pages to get prettier schema url?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +55,7 @@ Lassie can be controlled with a config file. To enable lassie for a project you
 
 ```json5
 {
+  "$schema": "https://raw.githubusercontent.com/geprog/lassie-bot-dog/main/config/lassie.schema.json",
```
Comment (anbraten) [inline README.md:58]: Using Github pages here should be easy:

```yml
name: Publish JSON schema
on:
  push:
    paths:
      - 'config/lassie.schema.json'
    branches:
      - main

jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout 🛎️
        uses: actions/checkout@v2.3.1

      - name: Remove all files except lassie.schema.json
      - run: |
         cd config
         rm -v !("lassie.schema.json") 

      - name: Deploy 🚀
        uses: JamesIves/github-pages-deploy-action@4.1.4
        with:
          branch: gh-pages
          folder: config
          clean: false
```

File: config/lassie.schema.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          "description": "Configuration for the auto_merge plugin",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "squash": {
+              "description": "Whether or not to squash the merge request when merging",
+              "type": "boolean"
+            },
+            "neededLabels": {
+              "description": "Labels needed for merge request being considered",
```
Comment (anbraten) [inline config/lassie.schema.json:25]:
```suggestion
              "description": "Labels needed for merge request being considered to be ready to auto merge",
```

@lukashass Maybe adding a new options like the one you described would be nice as well.

**Review by anbraten — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (anbraten) [reply README.md:58]: @lukashass You suggested to leave the url as is to keep it properly versioned.

I think we will always use  the latest anyways and as older versions would not be supported it should be fine to only release the current master version to github pages.

File: config/lassie.schema.json (hunk 2 above)
Comment (lukashass) [reply config/lassie.schema.json]: See #39

File: README.md (hunk 1 above)
Comment (lukashass) [reply README.md]: I added this workflow.

File: .github/workflows/publish-json-schema.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Publish JSON schema
+on:
+  push:
+    paths:
+      - 'config/lassie.schema.json'
```
Comment (anbraten) [inline .github/workflows/publish-json-schema.yml:5]: Nice idea.

File: README.md [hunk 4]
```diff
@@ -55,6 +55,7 @@ Lassie can be controlled with a config file. To enable lassie for a project you
 
 ```json5
 {
+  "$schema": "https://geprog.github.io/lassie-bot-dog/lassie.schema.json",
```
Comment (anbraten) [inline README.md:58]: @lukashass Seems to me that the schema is release under: `https://geprog.github.io/lassie-bot-dog/config/lassie.schema.json`

**Review by anbraten — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (lukashass) [reply README.md:58]: @anbraten No. Have a look, its already there :wink: https://geprog.github.io/lassie-bot-dog/lassie.schema.json

**Review by anbraten — APPROVED** [review]

**Pull request merged**